### PR TITLE
Add "Contributing to repositories" best practices

### PR DESCRIPTION
**Pull request by uschi2000**

### CODE REVIEW

**Review by ghost — APPROVED** [review]

File: docs/best-practices/contributing/readme.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+with performance impact, additional dependencies, etc.) mandate prior discussion. This conversation is a great
+opportunity to discuss how the proposed change aligns with the team's timeline and goals, whether its architecture and
+implementation is sound, whether someone else is already working on the same issue, etc.
+
+
+### Separate refactoring steps from actual contributions
+Though this is general recommendation for all code changes, contributions to other teams' repositories in particular
+should not conflate feature/fix work with unrelated refactoring work. If your contribution requires a refactoring step,
+it is often a good idea to first submit a refactoring-only contribution, and then a follow-up contribution with the
+feature or fix.
```
Comment (ellisjoe) [inline docs/best-practices/contributing/readme.md:29]: separating feature work from wiring the feature work in is also a useful way to break a change apart

Comment (j-baker) [reply docs/best-practices/contributing/readme.md:29]: yeah definitely - makes it way easier to contribute and generally makes the feature actually able to be added incrementally.

Comment (uschi2000) [reply docs/best-practices/contributing/readme.md:29]: added

File: docs/best-practices/contributing/readme.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+This section summarizes best practices for contributing to code repositories.
+
+*TL;DR: Reach out to the repository owners before making a pull-request, keep contributions focussed.*
+
+### Discuss the scope of your contribution
+Whether contributing to a repository owned by your team, or to another team's product or library, the first step should
+almost always (see below for exceptions) be to reach out to the team and discuss your contribution:
+
+- Check the relevant GitHub or JIRA issues for related work
```
Comment (markelliot) [inline docs/best-practices/contributing/readme.md]: probably also want to look for a contributions file that describes specializations for a particular project

File: docs/best-practices/contributing/readme.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+feature or fix.
+
+
+### Match existing code style
+Repositories owned by other teams may not always follow your preferred code style. Try to match the existing code style
+and avoid refactoring/restyling existing code unless necessary for your change (cf., [Separate refactoring and other
+contributions](#separate-refactoring-and-other-contributions)).
+
+
+### Own the contribution until it is merged
```
Comment (markelliot) [inline docs/best-practices/contributing/readme.md:38]: reading this think maybe it's useful to discuss the receiver's responsibilities, too

File: docs/best-practices/contributing/readme.md (hunk 2 above)
Comment (uschi2000) [reply docs/best-practices/contributing/readme.md]: done

File: docs/best-practices/contributing/readme.md (hunk 3 above)
Comment (uschi2000) [reply docs/best-practices/contributing/readme.md:38]: done

**Pull request merged**